docs/streaming/transport.mdx

```diff
@@ -0,0 +1,189 @@
+---
+title: "Transport"
+description: "A deep dive into the transport layer for streaming agents"
+---
+
+AgentKit's UI streaming is designed to be transport-agnostic, giving you the flexibility to use different real-time communication strategies. At its core, AgentKit doesn't manage WebSocket connections or push data to clients directly.
+
+Instead, it provides a powerful `streaming` configuration hook where you provide a `publish` function. AgentKit calls this function with structured data chunks as your agent network executes, and you decide how to send that data to the client.
+
+This decoupled design means you can use any streaming provider that fits your needs.
+
+## Default Transport: Inngest Realtime
+
+For most applications, we recommend using **Inngest Realtime** as the transport layer. It's robust, scalable, and integrates seamlessly with Inngest's durable function execution, providing a resilient and reliable streaming experience.
+
+### How It Works
+
+The data flow with Inngest Realtime involves a few key components on your backend:
+
+1.  **Chat Route (`/api/chat/route.ts`)**: A standard API endpoint that receives a message from your UI and sends an event to Inngest to trigger an agent run.
+2.  **Token Route (`/api/realtime/token/route.ts`)**: A secure endpoint that generates a short-lived subscription token for the client. The client uses this token to connect to a specific Inngest Realtime channel.
+3.  **Inngest Function**: The function that runs your agent or network. You pass Inngest's `publish` function to AgentKit's `streaming.publish` hook. AgentKit generates the events, and Inngest handles the delivery to the subscribed client.
```

Comment:
I can't find where we document the `publish` callback that it requires Inngest realtime - which we should link back to. I see it in the usage guide or the example code, but not sure how someone is going to use how to define channels etc